src/Microsoft.Health.Fhir.Core/Features/Operations/Validate/ValidateOperationHandler.cs

```diff
@@ -5,21 +5,39 @@
 
 using System.Threading;
 using System.Threading.Tasks;
+using EnsureThat;
 using MediatR;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Features.Security.Authorization;
 using Microsoft.Health.Fhir.Core.Messages.Operation;
 using Microsoft.Health.Fhir.Core.Models;
 
 namespace Microsoft.Health.Fhir.Core.Features.Validation
 {
     public class ValidateOperationHandler : IRequestHandler<ValidateOperationRequest, ValidateOperationResponse>
     {
+        private readonly IFhirAuthorizationService _authorizationService;
+
+        public ValidateOperationHandler(IFhirAuthorizationService authorizationService)
+        {
+            EnsureArg.IsNotNull(authorizationService, nameof(authorizationService));
+
+            _authorizationService = authorizationService;
+        }
+
         /// <summary>
         /// Handles validation requests that produced no errors. All validation is preformed before this is called.
         /// </summary>
         /// <param name="request">The request</param>
         /// <param name="cancellationToken">The CancellationToken</param>
         public Task<ValidateOperationResponse> Handle(ValidateOperationRequest request, CancellationToken cancellationToken)
         {
+            if (_authorizationService.CheckAccess(ResourceActions.ResourceValidate) != ResourceActions.ResourceValidate)
```

Comment:
I really like that we're moving these checks into the handlers. A possible enhancement might be to have the check access logic at a pre-processor level. Similar to the checks we do for validation and conformance (https://github.com/microsoft/fhir-server/blob/master/src/Microsoft.Health.Fhir.Core/Features/Validation/ValidateCapabilityPreProcessor.cs).

If the ResourceActions are static perhaps we could attribute the messages?

Reply:
(PR is still a draft)
They are not entirely static. For instance, for a u PUT you must have `update` but `create` is optional. Also, for bundles and transactions, I'd like to perform one check for all actions that are actually needed (but I'm not sure we have that capability yet). 
But we could do the preprocessor if we pass the response to the handler. But then again, when we implement slices, we'll need more from the auth service (to determine if a create is within a slice for instance)